src/main/java/com/gradle/enterprise/conventions/customvalueprovider/GitInformationCustomValueProvider.java

```diff
@@ -17,7 +18,11 @@ public GitInformationCustomValueProvider(GradleEnterpriseConventions conventions
     @Override
     public void accept(Settings settings, BuildScanExtension buildScan) {
         File rootDir = settings.getRootDir();
-        buildScan.background(__ -> {
+        buildScan.background(logGitInformationInBackground(rootDir));
+    }
+
+    private static Action<BuildScanExtension> logGitInformationInBackground(File rootDir) {
```

Comment:
🤔 I suppose I'd use a static inner class extending `Action` here instead of casting a lambda to on action in the method. Is there any advantage in using the lambda?

Reply:
> 🤔 I suppose I'd use a static inner class extending `Action` here instead of casting a lambda to on action in the method. Is there any advantage in using the lambda?

Less boilerplate, as usual. Otherwise there has to be a full class declaration with a constructor for `rootDir` capture.

Reply:
As long as the configuration cache can serialize that, all good.